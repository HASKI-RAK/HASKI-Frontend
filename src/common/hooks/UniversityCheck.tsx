import log from 'loglevel'
import { usePersistedStore } from '@store'
import { useState, useEffect, useMemo } from 'react'

export type UniversityHookReturn = {
  readonly university: string
}

export const useUniversity = (): UniversityHookReturn => {
  const [university, setUniversity] = useState('')
  const getUser = usePersistedStore((state) => state.getUser)
  useEffect(() => {
    //fetch the university from the current user and return university
    getUser().then((user)=> {
          setUniversity(user.university)
        })
        .catch((reason) => {
          log.error(reason)
          setUniversity('')
        })
    },[])
  return useMemo(
    () => ({
      university
    }),
    [university]
  )
}
