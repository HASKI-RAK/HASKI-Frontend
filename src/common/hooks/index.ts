export { useTheme } from './DefaultUseTheme/DefaultUseTheme'
export { useMediaQuery } from './DefaultMediaQuery/DefaultMediaQuery'
