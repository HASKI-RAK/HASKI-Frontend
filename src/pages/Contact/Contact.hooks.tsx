import log from 'loglevel'
import { useCallback, useContext } from 'react'
import { useTranslation } from 'react-i18next'
import { FormDataType, SnackbarContext, postContactForm } from '@services'
import { usePersistedStore } from '@store'

export type ContactHookProps = {
  setIsLoading: (isLoading: boolean) => void
}

export type ContactHookReturn = {
  /** handles the submission of the values, that were inserted in the Contactform. */
  onSubmitHandler: (postBody: FormDataType) => void
}
/**
 * Hook for the Contact logic. Handles reporttype, reporttopic and description state which sets the input for the textfields and
 * provides function to submit the form. Should usually override the submit function in the ContactForm.hooks
 *
 * @param props - {@link ContactHookProps}
 * @returns - {@link ContactHookReturn.onSubmitHandler} - Handles the submission of the values, that were inserted in the Contactform.
 * @remarks
 * Prior to sending with the postContactForm function the userid is getting fetched so that it can be sent with the postbody to the backend.
 * It posts the content thats supposed to be sent with the {@link FormDataType | postBody} with the postContactForm Function (can be found {@link postContactForm | here})
 * Upon recieving a response the submission was either successful, successful but still failed oder threw an error.
 * Supports usage of the Snackbar.
 *
 * The {@link FormDataType | postBody} is the object that will be sent to the backend.
 */

export const useContact = ({ setIsLoading }: ContactHookProps): ContactHookReturn => {
  const { t } = useTranslation()
  const { addSnackbar } = useContext(SnackbarContext)
  const getUser = usePersistedStore((state) => state.getUser)
  const onSubmitHandler = useCallback(
    (postBody: FormDataType) => {
      setIsLoading(true)
      getUser()
        .then((user) => {
          postContactForm(postBody, user.settings.user_id, user.lms_user_id)
            .then(() => {
              setIsLoading(false)
              addSnackbar({
                message: t('pages.contact.success'),
                severity: 'success'
              })
            })
            .catch((error) => {
              setIsLoading(false)
              addSnackbar({
                message: t('pages.contact.error'),
                severity: 'error',
                autoHideDuration: 3000
              })
              log.error(t('pages.contact.error') + ' ' + error)
            })
        })
        .catch((error) => {
          setIsLoading(false)
          addSnackbar({
            message: t('pages.contact.error'),
            severity: 'error',
            autoHideDuration: 3000
          })
          log.error(t('pages.contact.error') + ' ' + error)
        })
    },
    [t, addSnackbar, setIsLoading]
  )

  return {
    onSubmitHandler
  } as const
}
