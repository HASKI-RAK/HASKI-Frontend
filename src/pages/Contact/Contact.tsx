import { Contactform } from "@components";

//if a skeleton component is needed for the contact page add code below
/*function Contactskeleton(props: { loading?: boolean }) {
  const { loading = false } = props;
  return (
    
    <div>
      <Box sx={{ width: '50%', height: 400 }}>
      {loading?(
        <Skeleton id="loadingSkeleton_contact" variant="rectangular" width='100%' height='100%'/>
      ):(
        <Contactform width='100%'/>
        
      )}
      </Box>
    </div>
  );
}*/

export const SendToBackend = () => {
  console.log("send to backend");
};

export const Contact = () => {

  return (
    <div>
      
      <Contactform width="50%" sendtoBackend={SendToBackend}/>
      
    </div>
  )
}
export default Contact; 