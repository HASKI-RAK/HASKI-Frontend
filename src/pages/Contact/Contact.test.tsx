import { Contact } from '@pages'
import '@testing-library/jest-dom'
import { render, fireEvent, act, renderHook } from '@testing-library/react'
import { ContactForm } from '@components'
import { FormDataType, SnackbarContext, SnackbarContextType } from '@services'
import { useContact } from './Contact.hooks'
import { mockServices } from 'jest.setup'
import { MemoryRouter } from 'react-router-dom'

/*jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useCallback: (a: any) => a
}))*/

const mockSnackbarContext: SnackbarContextType = {
  snackbarsErrorWarning: [],
  snackbarsSuccessInfo: [],
  setSnackbarsErrorWarning: (a: any[]) => {
    return a
  },
  setSnackbarsSuccessInfo: (a: any) => {
    return a
  },
  addSnackbar: (a: any) => {
    return a
  },
  updateSnackbar: (a: any) => {
    return a
  },
  removeSnackbar: (a: any) => {
    return a
  }
}

/** 
 * useTranslation mocks the translation and also mocks the map input of reportTypes and topics,
 * input isnt important here.
 * global.fetch mocks the fetch function, which is used in the onSubmitHandler function in Contact.hooks.tsx
 * useContact mocks the useContact function, which is used in the Contact.tsx
 * mockSnackbarContext is a mocked Snackbarcontext, which handles the testing of the snackbar, when a form is submitted
 *
 * Currently getting tested are the cases:
 *
 * - Contactform is not getting sent at all. Only renders the form
 * - a form gets submitted, so that the fetch function gets called and returns 201
 * - tests a normal submit with filled in textfields, that are required
 * - a form gets submitted, but throws an error
 *
 * - the fetch function returns 201, but is Loading is false, so the Snackbar shows an error
 */
describe('Test Contactpage', () => {
  const submit = jest.fn()
  const useContact = jest.fn(() => {
    return { onSubmitHandler: submit }
  })

  test('not sending', () => {
    render(
      <SnackbarContext.Provider value={mockSnackbarContext}>
        <Contact />
      </SnackbarContext.Provider>
    )
    expect(useContact).not.toBeCalled()
  })
  test('sends onSubmit to Contactform', () => {
    const form = render(
      <MemoryRouter>
        <SnackbarContext.Provider value={mockSnackbarContext}>
          <ContactForm onSubmit={useContact} />
        </SnackbarContext.Provider>
      </MemoryRouter>
    )

    const submitButton = form.getByText('components.ContactForm.submit')
    const input = form.getByRole('textbox')
    fireEvent.change(input, { target: { value: 'text' } })
    fireEvent.mouseDown(form.getByRole('button', { name: /Topic/i }))
    act(() => {
      form.getAllByRole('option')[0].click()
    })
    fireEvent.click(submitButton)
    expect(useContact).toBeCalled()

    render(<Contact />)
  })

  test('test catch error', async () => {
    global.fetch = jest.fn(() =>
      Promise.resolve({
        json: () => {
          throw new Error('Error')
        }
      })
    ) as jest.Mock
    const result = await fetch(process.env.BACKEND + `/contactform`)
    expect(result.status).toBe(undefined)
  })
})

describe('Test on submit Function', () => {
  const testData: FormDataType = {
    report_type: '1',
    report_topic: '1',
    report_description: 'test'
  }
  test('Fetch successful', async () => {
    const loadingMock = jest.fn()
    const addSnackbarMock = jest.fn()

    const my_context = {
      snackbarsErrorWarning: [],
      snackbarsSuccessInfo: [],
      setSnackbarsErrorWarning: (a: any[]) => a,
      setSnackbarsSuccessInfo: (a: any) => a,
      addSnackbar: (a: any) => {
        addSnackbarMock(a)
        return a
      },
      updateSnackbar: (a: any) => a,
      removeSnackbar: (a: any) => a
    }

    const result = renderHook(
      () => {
        const onSubmit = useContact({ setIsLoading: loadingMock })
        return onSubmit
      },
      { wrapper: ({ children }) => <SnackbarContext.Provider value={my_context}>{children}</SnackbarContext.Provider> }
    )

    const onSubmit = result.result.current
    await act(async () => {
      onSubmit.onSubmitHandler(testData)

      // Check if loading is set True
      expect(loadingMock).lastCalledWith(true)

      // wait for all async calls
      await Promise.resolve()
    })

    expect(addSnackbarMock.mock.lastCall[0].severity).toEqual('success')
    expect(loadingMock).lastCalledWith(false)
  })

  test('Fetch fails', async () => {
    const loadingMock = jest.fn()
    const addSnackbarMock = jest.fn()

    const my_context = {
      snackbarsErrorWarning: [],
      snackbarsSuccessInfo: [],
      setSnackbarsErrorWarning: (a: any[]) => a,
      setSnackbarsSuccessInfo: (a: any) => a,
      addSnackbar: (a: any) => {
        addSnackbarMock(a)
        return a
      },
      updateSnackbar: (a: any) => a,
      removeSnackbar: (a: any) => a
    }

    const result = renderHook(
      () => {
        const onSubmit = useContact({ setIsLoading: loadingMock })
        return onSubmit
      },
      { wrapper: ({ children }) => <SnackbarContext.Provider value={my_context}>{children}</SnackbarContext.Provider> }
    )

    const onSubmit = result.result.current

    mockServices.getUser = jest.fn().mockImplementationOnce(() => 
     Promise.reject(new Error('get User failed')))
    
    await act(async () => {
      onSubmit.onSubmitHandler(testData)

      // Check if loading is set True
      expect(loadingMock).lastCalledWith(true)

      // wait for all async calls
      await Promise.resolve()
    })

    expect(addSnackbarMock.mock.lastCall[0].severity).toEqual('error')
    expect(loadingMock).lastCalledWith(true)
  })

  test('Fetch throws an error, Snackbar error', async () => {
    /*global.fetch = jest.fn(() => {
      throw new Error('Error')
    }) as jest.Mock*/

    mockServices.postContactForm = jest.fn().mockImplementationOnce(() => Promise.reject(new Error('error')))
    const loadingMock = jest.fn()
    const addSnackbarMock = jest.fn()

    const my_context = {
      snackbarsErrorWarning: [],
      snackbarsSuccessInfo: [],
      setSnackbarsErrorWarning: (a: any[]) => a,
      setSnackbarsSuccessInfo: (a: any) => a,
      addSnackbar: (a: any) => {
        addSnackbarMock(a)
        return a
      },
      updateSnackbar: (a: any) => a,
      removeSnackbar: (a: any) => a
    }

    const result = renderHook(
      () => {
        const onSubmit = useContact({ setIsLoading: loadingMock })
        return onSubmit
      },
      { wrapper: ({ children }) => <SnackbarContext.Provider value={my_context}>{children}</SnackbarContext.Provider> }
    )

    const onSubmit = result.result.current
    await act(async () => {
      onSubmit.onSubmitHandler(testData)

      await Promise.resolve()
    })

    expect(addSnackbarMock.mock.lastCall[0].severity).toEqual('error')
    expect(loadingMock).lastCalledWith(true)
  })
})
