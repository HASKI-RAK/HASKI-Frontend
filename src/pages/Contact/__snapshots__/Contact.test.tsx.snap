// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and matches snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="css-16v5qpr-MuiStack-root"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        components.contactform.contactform
      </h5>
      <div
        className="css-1p5q5e5-MuiStack-root"
      >
        <div
          className="MuiFormControl-root css-1otb2k4-MuiFormControl-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={true}
            id="select_label_contact"
          >
            components.contactform.topic
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            onClick={[Function]}
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="select_label_contact mui-component-select-reporttopic"
              className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-component-select-reporttopic"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              components.contactform.other
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              disabled={false}
              name="reporttopic"
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value="other"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-14lo706"
              >
                <span>
                  topic
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            className="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
            id="radio_contact_label"
          >
            components.contactform.reportType
          </label>
          <div
            className="MuiFormGroup-root MuiFormGroup-row css-qfz70r-MuiFormGroup-root"
            role="radiogroup"
          >
            <label
              className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                className="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <input
                  checked={false}
                  className="PrivateSwitchBase-input css-1m9pwf3"
                  disabled={false}
                  name="reporttype"
                  onChange={[Function]}
                  required={false}
                  type="radio"
                  value="issue"
                />
                <span
                  className="css-hyxlzm"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                    data-testid="RadioButtonUncheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                    data-testid="RadioButtonCheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </span>
              </span>
              <span
                className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              >
                components.contactform.issue
              </span>
            </label>
            <label
              className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                className="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <input
                  checked={false}
                  className="PrivateSwitchBase-input css-1m9pwf3"
                  disabled={false}
                  name="reporttype"
                  onChange={[Function]}
                  required={false}
                  type="radio"
                  value="bug"
                />
                <span
                  className="css-hyxlzm"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                    data-testid="RadioButtonUncheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                    data-testid="RadioButtonCheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </span>
              </span>
              <span
                className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              >
                components.contactform.bug
              </span>
            </label>
            <label
              className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                className="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <input
                  checked={false}
                  className="PrivateSwitchBase-input css-1m9pwf3"
                  disabled={false}
                  name="reporttype"
                  onChange={[Function]}
                  required={false}
                  type="radio"
                  value="feedback"
                />
                <span
                  className="css-hyxlzm"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                    data-testid="RadioButtonUncheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                    data-testid="RadioButtonCheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </span>
              </span>
              <span
                className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              >
                components.contactform.feedback
              </span>
            </label>
            <label
              className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                className="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <input
                  checked={false}
                  className="PrivateSwitchBase-input css-1m9pwf3"
                  disabled={false}
                  name="reporttype"
                  onChange={[Function]}
                  required={false}
                  type="radio"
                  value="feature"
                />
                <span
                  className="css-hyxlzm"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                    data-testid="RadioButtonUncheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                    data-testid="RadioButtonCheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </span>
              </span>
              <span
                className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              >
                components.contactform.feature
              </span>
            </label>
            <label
              className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                className="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary Mui-checked MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <input
                  checked={true}
                  className="PrivateSwitchBase-input css-1m9pwf3"
                  disabled={false}
                  name="reporttype"
                  onChange={[Function]}
                  required={false}
                  type="radio"
                  value="other"
                />
                <span
                  className="css-hyxlzm"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                    data-testid="RadioButtonUncheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
                    data-testid="RadioButtonCheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </span>
              </span>
              <span
                className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              >
                components.contactform.other
              </span>
            </label>
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          data-testid="desc_input"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="desc_input"
            id="desc_input-label"
          >
            components.contactform.briefDescription
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="desc_input"
              name="description"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              rows={5}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  components.contactform.briefDescription
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1pvw1xr-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          components.contactform.submit
        </button>
      </div>
    </div>
  </form>
</div>
`;
